### ISSUE TITLE
top left slide disappear when click right arrow

### ISSUE THREAD

**Issue by raganello1** [bqworks/slider-pro#238]
Hi and thanks for this, it's great.
I only have an issue, using a carousel layout, full width, when scrolling to the right the top left image instantly disappear (it goes to the top right becoming invisible to the viewport) before the slider animated to the left.
I checked from the inspector that simply the "translate3d(-465px, 0px, 0px)" css property push right way the image to top right of the slider but it should slide to the left out of the layou.

Any css or js workaround?

Thanks a lot!!

**Comment by raganello1** [comment]
UPDATE
I actually noticed that this stop happening if slides are more than 4. Starting from 5 this bug disappear, it should work even with 4 as the slider would be with slides 1,2,3 on viewport and 4 out so the number 1 doesn't need to jump before the 4 and could slide left out of the viewport.

How to achieve that? I mean how to have slide on top left sliding on left direction out of viewport even with 4 slides total? as for now it disappear and jump to top right slider out of viewport.

thanks!

**Comment by davidghi** [comment]
Hi

This is actually expected behaviour and it was discussed before: https://github.com/bqworks/slider-pro/issues/238

It might not be ideal for your particular case (which is rare, looping through only four items out of which three are visible), but it's better for several reasons. Since `loop` is enabled, at one point slide 1 needs to be positioned after slide 4. If the change takes place after the sliding is over (the behaviour which would be preferable for your case), there is a potential problems which I wanted to avoid: if you don't slide to the next slide, but jump over two or three slides, to positions where there are no slides yet, you will get to an empty space and only after the animation is over, the slides will appear in place. I consider it's not good UX to have the slider animate to a blank space. It's even more problematic if you swipe faster through the slides, swiping to a new slide before the previous animation was over (and thus the slide positioning taking place). Swiping like this, you will soon swipe through blank space, because the slides will appear only when you leave a sliding animation to run completely.

I know the behaviour is not ideal for your case, but, in my opinion, for that case it's better to disable `loop`.

Cheers!

**Comment by raganello1** [comment]
Thanks a lot for this huge help!

I understand what you're saying and how it's working but today I found a workaround which might be helpful for those in the same situation. 
I simply doubled the number of slides putting the same content twice, so user won't notice it's 8 slides instead of 4 (or 6 instead of 3) and I won't have this "top left slide" behavior

This time coding is unnecessary, of course it's not the best way to do so but it works :)

THANKS!
